### ISSUE TITLE
Lightmap dynamic objects get lighting from invalid probes

### ISSUE THREAD

**Issue by jcostello** [godotengine/godot#77168]
### Godot version

v4.1.dev.custom_build [437041a98]

### System information

Ubuntu with Nvidia 2060

### Issue description

Dynamic objects get light from invalid probes

![Screenshot from 2023-05-17 15-17-47](https://github.com/godotengine/godot/assets/1168582/66e3b0a3-94cb-46b3-b50b-e475937d41a1)

### Steps to reproduce

Use MRP

Bake the lightmap (in medium quality)
Move the mesh in the editor to get the dynamic lighting.
You will get same result as the picture above

### Minimal reproduction project

[lightmap.zip](https://github.com/godotengine/godot/files/11501406/lightmap.zip)

**Comment by Calinou** [comment]
I don't see a way to resolve this easily. We can't perform a collision check (as that may not always make sense), and generating probes in solid geometry is *probably* intended behavior (as you'd lose a lot of detail otherwise).

**Comment by jcostello** [comment]
> I don't see a way to resolve this easily. We can't perform a collision check (as that may not always make sense), and generating probes in solid geometry is _probably_ intended behavior (as you'd lose a lot of detail otherwise).

Inside solid geometry probes will be black anyways because they wont get any light. So meshes close to solid geometry will get darker. I don't think that's appropriate. It would be nice at least a way to remove manually probes that are not desirable

**Comment by clayjohn** [comment]
Since https://github.com/godotengine/godot/issues/77169 is fixed by https://github.com/godotengine/godot/pull/82068, I guess this issue will be too?

**Comment by jcostello** [comment]
Yes it is. But it will be good in the future to mark probes as invalid if they are inside a geometry (or if they can capture any light)
